runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/stream/GrpcClientFactory.java

```diff
@@ -450,21 +465,49 @@ private void doAppData(
             assert replySeq <= replyAck + replyMax;
         }
 
-        private void doAppAbort(
+        private void doAppAbortDeferring(
+            long traceId,
+            long authorization,
+            String16FW grpcStatus)
+        {
+            this.grpcStatus = grpcStatus != null ? grpcStatus.asString() : null;
+            this.state = GrpcState.closingReply(state);
+
+            if (GrpcState.replyOpened(state))
```

Comment:
We are setting the state to closing; the next condition is checking if it is still open. Is my understanding correct?

Reply:
Yes, state is not a single value, it flips bits for different parts of the overall state, so both are true at the same time.